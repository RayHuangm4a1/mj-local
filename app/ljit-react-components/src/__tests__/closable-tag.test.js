import React from 'react';
import { shallow, mount, render, } from 'enzyme';
import ClosableTag from '../components/closable-tag';

describe('Closable Tag', () => {
	it('handle default props', () => {
		const {
			isWithOutline,
			onClose,
		} = ClosableTag.defaultProps;

		expect(isWithOutline).toEqual(true);
		expect(onClose).toBeDefined();
		expect(onClose).toBeInstanceOf(Function);
	});

	it('should renders correctly', () => {
		const wrapper = render(
			<ClosableTag
				className="mock-class"
				onClose={() => {}}
				text="tag text"
			/>
		);

		expect(wrapper).toMatchSnapshot();
	});

	it('should be selectable by class "ljit-closable-tag"', () => {
		const wrapper = shallow(<ClosableTag />);

		expect(wrapper.hasClass('ljit-closable-tag')).toEqual(true);
	});

	it('should be selectable by custom class', () => {
		const className = 'mock-class';
		const wrapper = shallow(<ClosableTag className={className} />);

		expect(wrapper.hasClass(className)).toEqual(true);
	});

	describe('when tag is closable', () => {
		it('should be selectable by "ljit-closable-tag--closed"', () => {
			const wrapper = shallow(<ClosableTag />);

			expect(wrapper.hasClass('ljit-closable-tag--closed')).toEqual(true);
		});
	});

	it('should mount in a full DOM', () => {
		const text = 'mock-text';
		const className = 'mock-class';
		const onClose = () => {};

		const wrapper = mount(
			<ClosableTag
				text={text}
				className={className}
				onClose={onClose}
			/>
		);

		expect(wrapper.props().text).toEqual(text);
		expect(wrapper.props().className).toEqual(className);
		expect(wrapper.props().onClose).toEqual(onClose);
	});
});
