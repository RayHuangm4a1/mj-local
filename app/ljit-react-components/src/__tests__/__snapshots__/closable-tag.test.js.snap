// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Closable Tag should renders correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "ant-tag ljit-tag ljit-tag-shape--rectangle ljit-tag--border ljit-closable-tag ljit-closable-tag--closed mock-class",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "ljit-tag__text",
        },
        "children": Array [
          Object {
            "data": "tag text",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "span",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "aria-label": "icon: close",
            "class": "anticon anticon-close",
            "tabindex": "-1",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "aria-hidden": "true",
                "class": "",
                "data-icon": "close",
                "fill": "currentColor",
                "focusable": "false",
                "height": "1em",
                "viewBox": "64 64 896 896",
                "width": "1em",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "d": "M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z",
                  },
                  "children": Array [],
                  "name": "path",
                  "namespace": "http://www.w3.org/2000/svg",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "d": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "d": undefined,
                  },
                },
              ],
              "name": "svg",
              "namespace": "http://www.w3.org/2000/svg",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "aria-hidden": undefined,
                "class": undefined,
                "data-icon": undefined,
                "fill": undefined,
                "focusable": undefined,
                "height": undefined,
                "viewBox": undefined,
                "width": undefined,
              },
              "x-attribsPrefix": Object {
                "aria-hidden": undefined,
                "class": undefined,
                "data-icon": undefined,
                "fill": undefined,
                "focusable": undefined,
                "height": undefined,
                "viewBox": undefined,
                "width": undefined,
              },
            },
          ],
          "name": "i",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "aria-label": undefined,
            "class": undefined,
            "tabindex": undefined,
          },
          "x-attribsPrefix": Object {
            "aria-label": undefined,
            "class": undefined,
            "tabindex": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Object {
        "attribs": Object {
          "aria-label": "icon: close",
          "class": "anticon anticon-close",
          "tabindex": "-1",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "aria-hidden": "true",
              "class": "",
              "data-icon": "close",
              "fill": "currentColor",
              "focusable": "false",
              "height": "1em",
              "viewBox": "64 64 896 896",
              "width": "1em",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "d": "M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z",
                },
                "children": Array [],
                "name": "path",
                "namespace": "http://www.w3.org/2000/svg",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "d": undefined,
                },
                "x-attribsPrefix": Object {
                  "d": undefined,
                },
              },
            ],
            "name": "svg",
            "namespace": "http://www.w3.org/2000/svg",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "aria-hidden": undefined,
              "class": undefined,
              "data-icon": undefined,
              "fill": undefined,
              "focusable": undefined,
              "height": undefined,
              "viewBox": undefined,
              "width": undefined,
            },
            "x-attribsPrefix": Object {
              "aria-hidden": undefined,
              "class": undefined,
              "data-icon": undefined,
              "fill": undefined,
              "focusable": undefined,
              "height": undefined,
              "viewBox": undefined,
              "width": undefined,
            },
          },
        ],
        "name": "i",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "ljit-tag__text",
          },
          "children": Array [
            Object {
              "data": "tag text",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "span",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "aria-label": undefined,
          "class": undefined,
          "tabindex": undefined,
        },
        "x-attribsPrefix": Object {
          "aria-label": undefined,
          "class": undefined,
          "tabindex": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
